pkg/neorpc/result/peers.go

```diff
@@ -98,3 +116,16 @@ func (p *Peer) UnmarshalJSON(data []byte) error {
 	}
 	return err
 }
+
+func parseHostPort(addr string) (string, uint16, error) {
+	host, portStr, err := net.SplitHostPort(addr)
+	if err != nil {
+		return "", 0, err
+	}
+	port, err := strconv.ParseUint(portStr, 10, 16)
+	if err != nil {
+		return "", 0, err
+	}
+
+	return host, uint16(port), nil
```

Comment:
Let's preserve the original logic and do not return an error in case of invalid port:
```suggestion
	port, _ := strconv.ParseUint(portStr, 10, 16)
	return host, uint16(port), nil
```

And probably this method deserves a comment that it returns zero port in case of improperly formatted port string.